test/verify/check-packagekit

```diff
@@ -630,14 +631,258 @@ class TestUpdatesSubscriptions(PackageCase):
+        assertType("all")
+        assertTimer("06:00")
+
+        # change day
+        b.click("#auto-update-day button")
+        b.wait_present("#auto-update-day .dropdown-menu li[data-data='thu'] a")
+        b.click("#auto-update-day .dropdown-menu li[data-data='thu'] a")
+        wait_pending()
+        b.wait_in_text("#auto-update-day button", "Thursday")
+        assertType("all")
+        assertTimer("06:00", "Thu")
+
+        # change time
+        b.click("#auto-update-time button")
+        b.wait_present("#auto-update-time .dropdown-menu li[data-data='21:00'] a")
+        b.click("#auto-update-time .dropdown-menu li[data-data='21:00'] a")
+        wait_pending()
+        b.wait_in_text("#auto-update-time button", "21:00")
+        assertType("all")
+        assertTimer("21:00", "Thu")
+
+        # change back to daily
+        b.click("#auto-update-day button")
+        b.wait_present("#auto-update-day .dropdown-menu li[data-data=''] a")
+        b.click("#auto-update-day .dropdown-menu li[data-data=''] a")
+        wait_pending()
+        b.wait_in_text("#auto-update-day button", "every day")
+        assertType("all")
+        assertTimer("21:00")
+
+        # disable
+        b.click("#automatic div.btn-onoff-ct label.active span")
+        b.wait_in_text("#automatic div.btn-onoff-ct label.active", "Off")
+        self.assertFalse(b.is_present("#auto-update-type"))
+        assertTimer(None)
+
+    def testWithAvailableUpdates(self):
+        b = self.browser
+        m = self.machine
+
+        self.createPackage("vanilla", "1.0", "1", install=True)
+        self.createPackage("vanilla", "1.0", "2")
+        self.enableRepo()
+
+        m.start_cockpit()
+        b.login_and_go("/updates")
+        b.wait_present("#available")
+
+        if not self.supported_backend:
+            self.assertFalse(b.is_present("#automatic"))
+            self.assertFalse(b.is_present("#auto-update-type"))
+            return
+
+        b.wait_present("#automatic")
+        b.wait_present("#automatic div.btn-onoff-ct label.active")
+        b.wait_in_text("#automatic div.btn-onoff-ct label.active", "Off")
+        self.assertFalse(b.is_present("#auto-update-type"))
+
+        # enable
+        b.click("#automatic div.btn-onoff-ct label.active span")
+        b.wait_in_text("#automatic div.btn-onoff-ct label.active", "On")
+        b.wait_present("#auto-update-type")
+
+        if self.backend == 'dnf':
+            self.checkUpgradeRebootDnf()
+        else:
+            raise NotImplementedError(self.backend)
+
+    def checkUpgradeRebootDnf(self):
+        """part of testWithAvailableUpdates() for dnf backend"""
+
+        m = self.machine
+
+        # dial down the random sleep to avoid the test having to wait 5 mins
+        m.execute("sed -i '/random_sleep/ s/=.*$/= 3/' /etc/dnf/automatic.conf")
+        # then manually start the upgrade job like the timer would
+        m.execute("systemctl start dnf-automatic-install.service")
```

Comment:
Add a `--wait`?

Reply:
No need, it's `Type=oneshot` thus the `start` will block until it's actually done. Also, I remember adding `--wait` not too long ago (late last year), so it might not yet be available on RHEL/CentOS 7.

Reply:
Roger.  Looks like you know a thing or two about systemd... :-)